handlers/handlers.go

```diff
@@ -129,41 +157,17 @@ func HasContentType(r *http.Request, mimetype string) bool {
 	return false
 }
 
-func processUploadVOD(url string) error {
-	// TODO: This function is only a scaffold for now
-
-	// TODO: Update hostnames and ports
-	mc := MistClient{apiUrl: "http://localhost:4242/api2", triggerCallback: "http://host.docker.internal:8080/api/mist/trigger"}
-
-	streamName := randomStreamName("catalyst_vod_")
-	if err := mc.AddStream(streamName, url); err != nil {
+func (d *CatalystAPIHandlersCollection) processUploadVOD(streamName, sourceURL, targetURL string) error {
+	if err := d.MistClient.AddStream(streamName, sourceURL); err != nil {
 		return err
 	}
-
-	// TODO: Move it to `Trigger()`
-	defer mc.DeleteStream(streamName)
-
-	if err := mc.AddTrigger(streamName, "PUSH_END"); err != nil {
+	if err := d.MistClient.AddTrigger(streamName, "PUSH_END"); err != nil {
 		return err
 	}
-	// TODO: Move it to `Trigger()`
-	defer mc.DeleteTrigger(streamName, "PUSH_END")
-
-	if err := mc.AddTrigger(streamName, "RECORDING_END"); err != nil {
+	if err := d.MistClient.PushStart(streamName, targetURL); err != nil {
```

Comment:
So will mist call the `catalyst-uploader` in order to save to that `targetURL`? Cause from what I understood, this will come from the `URL` inside the first `output_location` to have a `source_segments` output set to `true`, right?

The URLs in the `output_location` are actually Object Store URLs compatible with the `drivers` lib moved to [go-tools](https://github.com/livepeer/go-tools) from `go-livepeer`. This means that Mist won't be able to o a single `PUT` to that URL. The `catalyst-uploader` does have support for it tho, but from what I understood it is receiving not only the Object Store URL but the path of the file on the same URL. So someone would have to append the file paths to this Object Store URL for each segmented file that is saved. Do you know if mist is doing that for calling the `catalyst-api`?

Reply:
Good, question, calling `catalyst-uploader` from Mist is being implemented by @stronk-dev , so Marco may have some insights here.